src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -85,6 +85,11 @@ public override string ToString()
     }
 }";
 
+        private CSharpCompilation CreateNullableCompilation(string source)
+        {
+            return CreateCompilation(new[] { source }, options: WithNonNullTypesTrue());
+        }
```

Comment:
Consider inlining this method because we have avoided creating `CreateCompilation` wrappers so far. And the call can be simplified to `CreateCompilation(source, options: WithNonNullTypesTrue())`.

Reply:
We have `CreateCompilation` helpers for other feature test classes.  I honestly think we should migrate toward using the helper.  What if we decide to no longer have a compilation-wide option, and we have to prepend a directive?  This way there is one thing to change rather than thousands of tests.

Reply:
The feature-specific helper methods tend to be fairly rigid, with a limited set of options or none at all. As soon as a test requires additional options, we need to add  parameters to the helper or the test needs to call the underlying `CreateCompilation` method directly.

Calling `CreateCompilation` in each test makes it clear what the test relies on. Yes, there are some changes to the class of tests in a feature that could be encapsulated within custom helpers but those changes are less common.

---
In reply to: [272724892](https://github.com/dotnet/roslyn/pull/34781#discussion_r272724892) [](ancestors = 272724892)

Reply:
I'm not going to remove this helper in this PR, but I will not block its removal in the future.